diff --git a/scripts/build/binary_grub_cfg b/scripts/build/binary_grub_cfg
index c41a4fba4..f7c91b6f7 100755
--- a/scripts/build/binary_grub_cfg
+++ b/scripts/build/binary_grub_cfg
@@ -54,6 +54,30 @@ Grub_live_menu_entry() {
 
 	cat >> "${LIVE_ENTRIES_TMP}" <<END
 menuentry "${LABEL}"${HOTKEY} {
+	smbios -t 1 -s 5 --set=productname
+	if [ "\$productname" == "ROCK 5 Model B" ];then
+		devicetree $(dirname $KERNEL)/dtb/rockchip/rk3588-rock-5b.dtb
+	elif [ "\$productname" == "ROCK 5 Model A" ];then
+		devicetree $(dirname $KERNEL)/dtb/rockchip/rk3588s-rock-5a.dtb
+	elif [ "\$productname" == "Indiedroid Nova" ];then
+		devicetree $(dirname $KERNEL)/dtb/rockchip/rk3588s-9tripod-linux.dtb
+	elif [ "\$productname" == "NanoPC T6" ];then
+		devicetree $(dirname $KERNEL)/dtb/rockchip/rk3588-nanopc-t6.dtb
+	elif [ "\$productname" == "NanoPi R6S" ];then
+		devicetree $(dirname $KERNEL)/dtb/rockchip/rk3588s-nanopi-r6s.dtb
+	elif [ "\$productname" == "Edge2" ];then
+		devicetree $(dirname $KERNEL)/dtb/rockchip/rk3588s-khadas-edge2.dtb
+	elif [ "\$productname" == "R58 Mini" ];then
+		devicetree $(dirname $KERNEL)/dtb/rockchip/rk3588-blueberry-minipc-linux.dtb
+	elif [ "\$productname" == "R58X" ];then
+		devicetree $(dirname $KERNEL)/dtb/rockchip/rk3588-blueberry-edge-v10-linux.dtb
+	elif [ "\$productname" == "Orange Pi 5" ];then
+		devicetree $(dirname $KERNEL)/dtb/rockchip/rk3588s-orangepi-5.dtb
+	elif [ "\$productname" == "Orange Pi 5 Plus" ];then
+		devicetree $(dirname $KERNEL)/dtb/rockchip/rk3588-orangepi-5-plus.dtb
+	elif [ "\$productname" == "H88K" ];then
+		devicetree $(dirname $KERNEL)/dtb/rockchip/rk3588-hinlink-h88k.dtb
+	fi
 	linux	$KERNEL $APPEND
 	initrd	$INITRD
 }
@@ -114,7 +138,7 @@ fi
 
 # Default entries
 DEFAULT_FLAVOUR="$(echo ${LB_LINUX_FLAVOURS} | awk '{ print $1 }')"
-DEFAULT_KERNEL="$(basename chroot/boot/vmlinuz-*${DEFAULT_FLAVOUR})"
+DEFAULT_KERNEL="$(basename chroot/boot/vmlinuz-*)"
 DEFAULT_INITRD="initrd.img-$(echo ${DEFAULT_KERNEL} | sed -e 's|vmlinuz-||')"
 
 KERNEL_LIVE="/${INITFS}/${DEFAULT_KERNEL}"
