diff --git a/scripts/build/binary_grub_cfg b/scripts/build/binary_grub_cfg
index c41a4fba4..9d7fd0bdc 100755
--- a/scripts/build/binary_grub_cfg
+++ b/scripts/build/binary_grub_cfg
@@ -54,6 +54,12 @@ Grub_live_menu_entry() {
 
 	cat >> "${LIVE_ENTRIES_TMP}" <<END
 menuentry "${LABEL}"${HOTKEY} {
+	smbios -t 11 -s 4 --set=devicetreename
+	if [ -f $(dirname $KERNEL)/dtb/rockchip/\$devicetreename ];then
+		devicetree $(dirname $KERNEL)/dtb/rockchip/\$devicetreename
+	elif [ -f $(dirname $KERNEL)/dtb/rockchip/\$devicetreename.dtb ];then
+		devicetree $(dirname $KERNEL)/dtb/rockchip/\$devicetreename.dtb
+	fi
 	linux	$KERNEL $APPEND
 	initrd	$INITRD
 }
